frontend/www/js/omegaup/components/login/Signup.vue

```diff
@@ -106,31 +116,75 @@
 </template>
 
 <script lang="ts">
-import { Vue, Component, Prop } from 'vue-property-decorator';
+import { Vue, Component, Prop, Watch } from 'vue-property-decorator';
 import omegaup_Markdown from '../Markdown.vue';
 import T from '../../lang';
-
+import * as time from '../../time';
+import DatePicker from '../DatePicker.vue';
+import * as ui from '../../ui';
 @Component({
   components: {
     'omegaup-markdown': omegaup_Markdown,
+    'omegaup-datepicker': DatePicker,
   },
 })
 export default class Signup extends Vue {
   @Prop() validateRecaptcha!: boolean;
-
   T = T;
   username: string = '';
-  email: string = '';
+  email: null | string = null;
+  parentEmail: null | string = null;
   password: string = '';
   passwordConfirmation: string = '';
   recaptchaResponse: string = '';
-
+  birthDate: null | Date = new Date(0);
+  checked = false;
+  get loginEmailDescriptionText(): string {
+    return this.userAge > 13 ? T.loginEmail : T.loginEmailParent;
+  }
+  get userAge(): number {
+    if (!this.birthDate) {
+      return 0;
+    }
```

Comment:
```suggestion
  get userAge(): number | null {
    if (this.birthDate === null) {
      return null;
    }
```

otherwise folks that _were_ indeed born on midnight January 1st, 1970 wouldn't be able to create an account without their parent's consent!